### PR TITLE
Regroup advanced settings

### LINKED ISSUE #5413: Point users to use a configuration file from advanced settings
Currently users may still be using the advanced settings to build their docs, they should use a configuration file, now they can do everything from the web in the config file.

I was thinking in blocking the UI for new projects, but we can't do that, because that would be a problem for people moving to read the docs (they can't modify their docs). A long solution can be allowing to have a global configuration (codecov has this feature), and that can be merged with the configuration of each version (probably I can create another issue to discuss this more if we agree).

Anyway, going back to the original issue:

We already have something like this in the settings

screenshot_2019-03-06 edit project read the docs

I was thinking we could use the same pattern in the advanced settings

screenshot_2019-03-06 edit advanced project settings read the docs

But I think that first we should group the settings, we have global and per version settings, so putting the global settings first would be the idea (those are not present in the config file). Then we can list the per-version settings (available in the config file) but first showing a tip/link to the config file docs.

```
Advanced settings

[ ] Single version
[Default branch]
[Default version]
....

:warning: All the next options can be configured using a configuration file.

[Python interpreter]
[ ] Istall project
[Requirements file]
....

```

### PR DESCRIPTION
**Pull request by stsewd**
This is a continuation of #5459

Closes #5413 

It looks like this

Screenshot_2019-03-18 Edit Advanced Project Settings Read the Docs

I'm not sure about the texts, so any help is welcome.

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -226,10 +216,30 @@ class Meta:
...
+        help_text = render_to_string(
+            'projects/project_advanced_settings_helptext.html'
+        )
+        self.helper.layout = Layout(
+            Fieldset(
+                _("Global settings"),
+                *self.Meta.per_project_settings,
+            ),
+            Fieldset(
+                _("Other settings"),
```
Comment (stsewd) [inline readthedocs/projects/forms.py]: I don't like this kind of division, but these are not per version settings (at least in this page)

**Comment by stsewd** [comment]
I haven't tested it with the corporate site, first I want to know if we are ok with this :)

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% load i18n %}
+<p class="empty">
```
Comment (stsewd) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html]: Used the same patter that we have in settings https://readthedocs.org/dashboard/docs/edit/

**Comment by humitos** [comment]
I think we could adapt the note in the second group to say something like: "These settings can be defined in the YAML file. We strongly recommend that." or something like that inviting people to use the config file instead of the web.

**Comment by ericholscher** [comment]
> I think we could adapt the note in the second group to say something like: "These settings can be defined in the YAML file. We strongly recommend that." or something like that inviting people to use the config file instead of the web.

Agreed with @humitos -- the goal here is to push people to use YAML. We should also likely have logic here that shows if they already have a YAML config in their builds, and then puts a big warning that these settings won't do anything, if they do.

**Comment by stsewd** [comment]
> We should also likely have logic here that shows if they already have a YAML config in their builds, and then puts a big warning that these settings won't do anything, if they do.

But we'll be lying, these settings will still affect to the versions without a config file, we can't determinate exactly that sadly.

**Comment by stsewd** [comment]
But we can show that in the build page :), I'm creating another issue to track that

**Comment by ericholscher** [comment]
> But we'll be lying, these settings will still affect to the versions without a config file, we can't determinate exactly that sadly.

Sure, but we can detect if any versions use a config, and make the warning more pronounced. We could even presumably query all active versions and say "these settings will apply to 2 out of 6 versions)" if it isn't too heavy on the DB.

**Comment by agjohnson** [comment]
I think we should avoid making our UX for this view more complicated and instead frame these as "Default options" -- meaning, there is a path to overriding the options. Adding display to the view that adds information on which versions use which configuration option feels like we're putting emphasis on the wrong UI.

The thing we can do here is completely hide the options for new projects, instead pointing them to the config file for *all* their versions. But i think we're addressing that separately.

**Comment by stsewd** [comment]
> The thing we can do here is completely hide the options for new projects, instead pointing them to the config file for all their versions. But i think we're addressing that separately.

I put some thoughts about that in #5413

> I was thinking in blocking the UI for new projects, but we can't do that, because that would be a problem for people moving to read the docs (they can't modify their docs). A long solution can be allowing to have a global configuration (codecov has this feature), and that can be merged with the configuration of each version (probably I can create another issue to discuss this more if we agree).

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  {% blocktrans with config_file_link="https://docs.readthedocs.io/page/config-file" %}
```
Comment (davidfischer) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html]: I'd link directly to the v2 config file docs: https://docs.readthedocs.io/page/config-file/v2.html

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    These settings can be configured using a
+    <a href="{{ config_file_link }}">configuration file</a>.
+    That's the recommended way of set up your project.
```
Comment (davidfischer) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html]:
```suggestion
    That's the recommended way to set up your project.
```

File: readthedocs/templates/projects/project_advanced_settings_helptext.html (hunk 3 above)
Comment (davidfischer) [reply readthedocs/templates/projects/project_advanced_settings_helptext.html]: Also, you might consider adding `trimmed` to the `blocktrans` tag.

**Comment by stsewd** [comment]
Looks like we need to fix something in the styles of the corporate site :)

Screenshot_2019-03-21 Edit Advanced Project Settings - Read the Docs for Business

- No borders (maybe this is part of the style?)
- Submit button
- Help/dialog text looks :100:

File: readthedocs/projects/forms.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/projects/forms.py]: I liked Anthony's idea of "Default Settings" which implies they can be overwritten by the YAML, which is good. 
```suggestion
                _("Default settings"),
```

File: readthedocs/templates/projects/project_advanced.html [hunk 5]
```diff
@@ -13,7 +14,7 @@
 
 {% block project_edit_content %}
   <form method="post" action=".">{% csrf_token %}
-    {{ form.as_p }}
+    {% crispy form %}
     <p>
       <input style="display: inline;" type="submit" value="{% trans "Save" %}">
```
Comment (dojutsu-user) [inline readthedocs/templates/projects/project_advanced.html]: @stsewd 
`{% crispy form %}` renders
```html
<form>
  <!-- csrf token -->
  <!-- content -->
</form>
```
And we have wrapped this in outer `<form>` tag and it is finally rendering something like this.
Screenshot from 2019-03-27 21-50-05

and so the `Save` button, which is rendered outside `<form>` tag, is not working anymore. We need to fix this before merging this in `rel` branch

docs: https://django-crispy-forms.readthedocs.io/en/latest/crispy_tag_forms.html#fundamentals

Comment (stsewd) [reply readthedocs/templates/projects/project_advanced.html]: https://github.com/rtfd/readthedocs.org/pull/5544

**Pull request merged**